python_modules/dagster/dagster/_core/definitions/declarative_scheduling/slice_condition.py

```diff
@@ -32,9 +34,21 @@ def compute_slice(self, context: SchedulingConditionEvaluationContext) -> AssetS
 
 
 class InLatestTimeWindowCondition(SliceSchedulingCondition):
+    lookback_seconds: Optional[float] = None
```

Comment:
Ok. I think it is important to abstract serialization away from the user and do it in a principled way.

According to chatgpt:

```
In Python, the datetime.timedelta object represents a duration, the difference between two dates or times. Internally, a datetime.timedelta object stores its data in three main attributes:

days: An integer representing the number of days.
seconds: An integer representing the number of seconds (not including the days, hence always less than one day).
microseconds: An integer representing the number of microseconds (not including the seconds, hence always less than one second).
```

So I think you should confirm this and then handled serialization here precisely, rather than relying on a float respresentation of seconds.

Also the user-facing constructor should take a `timedelta` object, not seconds.

Reply:
Because this file moves around a lot in later PRs, I added this functionality to the top of the stack.

ChatGPT was correct in this case, but still a little bit of weirdness in representing things this way, as discussed on: https://github.com/dagster-io/dagster/pull/21615